facet-macros-emit/src/process_struct.rs

```diff
@@ -399,66 +399,70 @@ pub(crate) fn process_struct(parsed: Struct) -> TokenStream {
     // Add try_from_inner implementation for transparent types
     let try_from_inner_code = if ps.container.attrs.is_transparent() {
         if let Some(inner_field) = &inner_field {
-            // Transparent struct with one field
-            let inner_field_type = &inner_field.ty;
-            let bgp_without_bounds = ps.container.bgp.display_without_bounds();
+            if !inner_field.attrs.is_opaque() {
```

Comment:
currently compilation fails if we try add `#[facet(opaque)]` inside a `#[facet(transparent)]`, as the generated try_from code here fails.

we could make it compile by changing `#inner_field_type` to be `::facet::Opaque<#inner_field_type>`, but I'm not sure if it makes any sense? for custom de/ser we want the non-try-from path anyways.

Reply:
For DeSer we prefer the `try_from` vtables to be `None` so that `begin_inner` will hit the [branch that calls `begin_nth_field(0)`](https://github.com/facet-rs/facet/blob/ee0684e50b9faf5b05ab5bff22fe2912c4d3cbef/facet-reflect/src/partial/partial_api/option.rs#L177). then everything else "just works".

Since there should be no one using `#[facet(opaque)]` inside `#[facet(transparent)]` already, because it wouldn't compile before, I think it's safe to go this route, and if someone comes along with a use case for having the try_from implementations on their opaque types called, we can revisit. My current knowledge of how they're used is lacking though :see_no_evil:, so I might be just be missing something.